docs/architecture/ADR-004-interest-and-reserves.md

```diff
@@ -0,0 +1,258 @@
+# ADR 004: Borrow interest implementation and reserves
+
+## Changelog
+
+- October 14, 2021: Initial Draft (@toteki)
+- October 18, 2021: More Details on Interest (@toteki)
```

Comment:
just a tiny nitpick, changelog entries aren't required when updating the ADR in the same PR. Changelog entries are for changes that happen in future subsequent PRs ;-)